### PR TITLE
Add fix_path plugin

### PR DESCRIPTION
**Pull request by bueli**
This plugin sanitizes pathnames containing // (double slash) which will trigger [Empty path component found in input](https://github.com/frej/fast-export/issues/240) issue

### CODE REVIEW

File: plugins/fix_path/__init__.py
Comment (frej) [inline plugins/fix_path/__init__.py:8]: Why add commented out code?

File: plugins/fix_path/__init__.py
Comment (frej) [inline plugins/fix_path/__init__.py:11]: Why do this conditionally? The filename cannot be empty. Also, why scan the string twice?

File: plugins/fix_path/README.md
Comment (frej) [inline plugins/fix_path/README.md:3]: Sentences start with a capital letter

File: plugins/fix_path/__init__.py
Comment (frej) [inline plugins/fix_path/__init__.py:8]: Don't add commented out code

**Review by frej — CHANGES_REQUESTED** [review]
Please take a look at [what is a good commit message](https://chris.beams.io/posts/git-commit/) and amend your message accordingly. Make sure the subject starts with a capital letter and provide at least a one sentence description about what the plugin does in the body.

**Comment by bueli** [comment]
Reduced code and fixed commit message as you requested

File: plugins/fix_path/__init__.py
Comment (frej) [inline plugins/fix_path/__init__.py:8]: You have not answered the question of why you scan the string twice. As far as I can tell the `if '//' in filename` is completely redundant. Am I missing something?

**Review by frej — CHANGES_REQUESTED** [review]
Please squash your commits, there is no reason to preserve the development history of this small change for eternity.

Also make sure that the body of the commit message is wrapped at, at most 80, chars. The [previous linked](https://chris.beams.io/posts/git-commit/) recommends around 72.

Request a re-review when done.

File: plugins/fix_path/__init__.py
Comment (frej) [reply plugins/fix_path/__init__.py]: Forget the comment about compiling the regexp, I read the code hastily, sorry.

**Comment by bueli** [comment]
Hi Frej

Will squash soon.

I think your quality goals are very appropriate for production code.
But here we are in a deserted corner of an auxiliary tool ...
Why do you spend and demand so much effort?

You almost lost me, but now i want to understand where the energy is coming from ...

Cheers,
bue

**Comment by frej** [comment]
> But here we are in a deserted corner of an auxiliary tool ... Why do you spend and demand so much effort? You almost lost me, but now i want to understand where the energy is coming from

It's to minimize the maintenance effort, the clearer the patches, the simpler it is to maintain as I can figure out what a change is supposed to be doing. This is a real time saver, as with the declining popularity of Mercurial, I expect the frequency of maintenance actions to become more and more infrequent. The less time I have to spend to remember how something works, the happier I am.

As a maintainer, merging code is always a risk, if I merge a sloppy looking patch, will I have to maintain it in the future, are there hidden bugs? A sloppy looking patch does not really inspire confidence in the quality of the code, especially if the effort to make it "right" in the first place would have been minimal and only would have required a read-through on part of the contributor. Also the willingness to respond to review suggestions and questions is, I have found, a pretty good indication if the contributor will help me if unforeseen problems show up after the merge.

**Comment by frej** [comment]
For the third time, why do you check if `//` occurs in the string before running `filename.replace("//", "/")`, instead of running `replace()` directly? Checking the Python implementation there is no performance benefit from it, you'll end up scanning the string twice when the pattern `//` is in the string.

**Comment by frej** [comment]
No response to review comments in more than 90 days, closing without prejudice for inactivity.

**Pull request closed without merging**
